pyartifactory/objects/artifact.py

```diff
@@ -96,29 +97,32 @@ def deploy(
                 for file in files:
                     self.deploy(Path(f"{root}/{file}"), Path(f"{new_root}/{file}"))
         else:
-            if checksums:
-                # based on https://github.com/thenewnano/python-artifactory/commit/3c095f88f5e4153d1feffbf3769d038eb16b4016
+            if checksum_enabled:
                 artifact_check_sums = Checksums.generate(local_file)
-                headers = {
-                    "X-Checksum-Deploy": "true",
-                    "X-Checksum-Sha1": artifact_check_sums.sha1,
-                    "X-Checksum-Sha256": artifact_check_sums.sha256,
-                    "X-Checksum": artifact_check_sums.md5,
-                }
+                try:
+                    self._put(
+                        f"{artifact_path}",
```

Comment:
self._put(route=artifact_folder.as_posix(), ...) instead because` artifact_path` is Union[Path, str]

Reply:
The tests are failing, and I believe this is the cause